### PR TITLE
Add AWS CodeCommit upstream for git segment

### PR DESCRIPTION
**Pull request by kema-dev**
### Prerequisites

- [x] I have read and understood the [contributing guide][CONTRIBUTING.md].
- [x] The commit message follows the [conventional commits][cc] guidelines.
- [x] Tests for the changes have been added (for bug fixes / features).
- [x] Docs have been added/updated (for bug fixes / features).

### Description

Add AWS CodeCommit upstream for git segment, and fix a typo

### How

Adding a regex to match appropriate URL, and fix a typo

### Notes

I did not find a better NF icon, however I would be happy if someone finds one. I'm also getting a space after the icon, I do not know why, help would be appreciated in order to fix this !

[CONTRIBUTING.md]: https://github.com/JanDeDobbeleer/oh-my-posh/blob/main/CONTRIBUTING.md
[cc]: https://www.conventionalcommits.org/en/v1.0.0/#summary

### CODE REVIEW

File: src/segments/git.go [hunk 1]
```diff
@@ -93,12 +93,14 @@ const (
...
 	UpstreamIcons properties.Property = "upstream_icons"
-	// GithubIcon shows√ when upstream is github
+	// GithubIcon shows when upstream is github
 	GithubIcon properties.Property = "github_icon"
 	// BitbucketIcon shows  when upstream is bitbucket
 	BitbucketIcon properties.Property = "bitbucket_icon"
 	// AzureDevOpsIcon shows  when upstream is azure devops
 	AzureDevOpsIcon properties.Property = "azure_devops_icon"
+	// AwsCodecommitIcon shows  when upstream is aws codecommit
+	AwsCodecommitIcon properties.Property = "aws_codecommit_icon"
```
Comment (JanDeDobbeleer) [inline src/segments/git.go]:
```suggestion
	CodecommitIcon properties.Property = "codecommit_icon"
```

I feel this has a better ring to it, the remote prefix is also `codecommit` it seems.

**Review by JanDeDobbeleer — CHANGES_REQUESTED** [review]
One remark, I would rename the icon property. Great PR!

**Comment by kema-dev** [comment]
Name changed, also fixed the upstream detection regex for codecommit as it was kinda wrong

As you can see in the image, I still get a space after the icon, I don't know why, I think I followed the same principle as other icons, but still, I get one more space.

image
image

I found `nf-md-aws` icon which would be better, but it's code is `\udb83\ude0f` and does not appear to be valid from my formatter :shrug: 

Thanks for creating and maintaining this awesome project though !

**Comment by JanDeDobbeleer** [comment]
@kema-dev it's the GH icon which renders too big on your terminal. Not every terminal deals with these fonts the same way, and some icons became smaller with NF iterations.

**Comment by JanDeDobbeleer** [comment]
@all-contributors please add @kema-dev for design,doc

**Pull request merged**